metricflow/model/semantics/semantic_model_lookup.py

```diff
@@ -92,34 +92,23 @@ def get_dimension_from_semantic_model(
 
     def get_dimension(self, dimension_reference: DimensionReference) -> Dimension:
         """Retrieves a full dimension object by name."""
-        for dimension_source in self._dimension_index[dimension_reference]:
-            dimension = SemanticModelLookup.get_dimension_from_semantic_model(
-                semantic_model=dimension_source, dimension_reference=dimension_reference
+        semantic_models = self._dimension_index.get(dimension_reference)
+        if not semantic_models:
```

Comment:
could this be `if semantic_models is None`? or are we expecting `[]` as a return value?

Reply:
I intentionally left it as a falseness check! I don't EXPECT `[]` as a return value but if for some reason that were to happen this would still handle that, so it seemed preferable.